include/enotify.php

```diff
@@ -118,316 +109,234 @@ function notification($params)
-	}
-
-	if ($params['type'] == Notification\Type::COMMENT || $params['type'] == Notification\Type::TAG_SELF) {
-		if (Post\ThreadUser::getIgnored($parent_uri_id, $params['uid'])) {
-			Logger::info('Thread is ignored', ['parent' => $parent_id, 'parent-uri-id' => $parent_uri_id]);
-			return false;
-		}
+		case Notification\Type::COMMENT:
+			if (Post\ThreadUser::getIgnored($parent_uri_id, $params['uid'])) {
+				Logger::info('Thread is ignored', ['parent' => $parent_id, 'parent-uri-id' => $parent_uri_id]);
+				return false;
+			}
 
-		// if it's a post figure out who's post it is.
-		$item = null;
-		if ($params['otype'] === Notification\ObjectType::ITEM && $parent_id) {
 			$item = Post::selectFirstForUser($params['uid'], Item::ITEM_FIELDLIST, ['id' => $parent_id, 'deleted' => false]);
-		}
-
-		if (empty($item)) {
-			return false;
-		}
+			if (empty($item)) {
+				return false;
+			}
 
-		$item_post_type = Item::postType($item, $l10n);
+			$item_post_type = Item::postType($item, $l10n);
 
-		$content = Plaintext::getPost($item, 70);
-		if (!empty($content['text'])) {
-			$title = '"' . trim(str_replace("\n", " ", $content['text'])) . '"';
-		} else {
-			$title = '';
-		}
+			$content = Plaintext::getPost($item, 70);
+			if (!empty($content['text'])) {
+				$title = '"' . trim(str_replace("\n", " ", $content['text'])) . '"';
+			} else {
+				$title = '';
+			}
 
-		// First go for the general message
+			// First go for the general message
 
-		// "George Bull's post"
-		if (!empty($params['activity']['origin_comment'])) {
-			$message = $l10n->t('%1$s replied to you on %2$s\'s %3$s %4$s');
-		} elseif (!empty($params['activity']['explicit_tagged'])) {
-			$message = $l10n->t('%1$s tagged you on %2$s\'s %3$s %4$s');
-		} else {
+			// "George Bull's post"
 			$message = $l10n->t('%1$s commented on %2$s\'s %3$s %4$s');
-		}
-
-		$dest_str = sprintf($message, $params['source_name'], $item['author-name'], $item_post_type, $title);
-
-		// Then look for the special cases
+			$dest_str = sprintf($message, $params['source_name'], $item['author-name'], $item_post_type, $title);
 
-		// "your post"
-		if (!empty($params['activity']['origin_thread'])) {
-			if (!empty($params['activity']['origin_comment'])) {
-				$message = $l10n->t('%1$s replied to you on your %2$s %3$s');
-			} elseif (!empty($params['activity']['explicit_tagged'])) {
-				$message = $l10n->t('%1$s tagged you on your %2$s %3$s');
-			} else {
+			// "your post"
+			if ($item['wall']) {
 				$message = $l10n->t('%1$s commented on your %2$s %3$s');
-			}
-
-			$dest_str = sprintf($message, $params['source_name'], $item_post_type, $title);
-		// "their post"
-		} elseif ($item['author-link'] == $params['source_link']) {
-			if (!empty($params['activity']['origin_comment'])) {
-				$message = $l10n->t('%1$s replied to you on their %2$s %3$s');
-			} elseif (!empty($params['activity']['explicit_tagged'])) {
-				$message = $l10n->t('%1$s tagged you on their %2$s %3$s');
-			} else {
```

Comment:
Why did you remove these distinctions?

Reply:
This is handled in another part of the system now.